src/objects/InstancedMesh.js

```diff
@@ -21,15 +28,65 @@ InstancedMesh.prototype = Object.assign( Object.create( Mesh.prototype ), {
 
 	isInstancedMesh: true,
 
-	raycast: function () {},
+	getMatrixAt: function ( index, matrix ) {
+
+		matrix.fromArray( this.instanceMatrix.array, index * 16 );
+
+	},
+
+	raycast: function ( raycaster, intersects ) {
+
+		var matrixWorld = this.matrixWorld;
+		var raycastTimes = this.count;
+
+		_agentMesh.geometry = this.geometry;
+		_agentMesh.material = this.material;
+
+		if ( _agentMesh.material === undefined ) return;
+
+		for ( var instanceId = 0; instanceId < raycastTimes; instanceId ++ ) {
+
+			//Calculate the world matrix for each instance
+
+			this.getMatrixAt( instanceId, _instanceLocalMatrix );
+
+			_instanceWorldMatrix.multiplyMatrices( matrixWorld, _instanceLocalMatrix );
+
+
+			//The agent mesh represents this single instance
+
+			_agentMesh.matrix = _instanceWorldMatrix;
```

Comment:
Are you sure this line is necessary? I would assume copying the world matrix is sufficient.

Reply:
It's true that just copying the world matrix is enough. I just want to keep the logic consistent. The local matrix is consistent with the world matrix, but it doesn't seem necessary now.Thank you for your advice.

Reply:
In that case, I don't understand why you assign the local matrix but copy the world matrix one line later. Any reasons for this?

Reply:
I just don't want two matrix objects to refer to the same matrix object, so as to avoid the change of one object causing the change of another.